Core/TermsOfService.swift

```diff
@@ -20,125 +20,84 @@
+        
+        case a = "A", b = "B", c = "C", d = "D"
+        
     }
-
-    public init(classification: Classification?, score: Int, goodReasons: [String], badReasons: [String]) {
-        self.classification = classification
-        self.score = score
-        self.goodReasons = goodReasons
-        self.badReasons = badReasons
+    
+    struct Reasons: Decodable {
+        
+        let good: [String]?
+        let bad: [String]?
+        
     }
-
-    private enum CodingKeys: String, CodingKey {
-        case classification = "class"
+    
+    let classification: Classification?
+    let score: Int
+    let reasons: Reasons
+    
+    enum CodingKeys: String, CodingKey {
         case score
-        case match
-
-        enum Match: String, CodingKey {
-            case good
-            case bad
-        }
-    }
-
-    struct Lookups {
-
-        static let classificationAsPractices: [Classification: PrivacyPractices] = [
-            .a: .good,
-            .b: .mixed,
-            .c: .poor,
-            .d: .poor,
-            .e: .poor
-            ]
-
-        static let classificationScores: [Classification: Int] = [
-            .a: -1,
-            .b: 0,
-            .c: 0,
-            .d: 1,
-            .e: 2
-            ]
-
-        static let derivedScoreAsPractices: [Int: PrivacyPractices] = [
-            -1: .good,
-            0: .mixed,
-            1: .poor
-            ]
-
-    }
-
-    public let classification: Classification?
-    public let score: Int
-    public let goodReasons: [String]
-    public let badReasons: [String]
-
-    public var hasReasons: Bool {
-        return !goodReasons.isEmpty || !badReasons.isEmpty
-    }
-
-    public var hasUnknownPractices: Bool {
-        return !hasReasons
+        case reasons = "match"
+        case classification = "class"
     }
-
-    public var derivedScore: Int {
+    
+    // see https://github.com/duckduckgo/duckduckgo-privacy-extension/blob/e42533/shared/js/background/privacy-practices.es6.js#L65
+    var summary: PrivacyPractices.Summary {
+        
         if let classification = classification {
-            return Lookups.classificationScores[classification]!
+            return TermsOfService.classificationSummaries[classification]!
         }
-
-        return normalizeScore()
-    }
-
-    public func privacyPractices() -> PrivacyPractices {
-        guard !hasUnknownPractices else { return .unknown }
-
-        var practices: PrivacyPractices?
-        if let classification = classification {
-            practices = Lookups.classificationAsPractices[classification]
-        } else if !goodReasons.isEmpty && !badReasons.isEmpty {
+        
+        if hasGoodReasons && hasBadReasons {
             return .mixed
-        } else {
-            practices = Lookups.derivedScoreAsPractices[normalizeScore()]!
         }
-
-        if let practices = practices {
-            return practices
+        
+        if score < 0 {
```

Comment:
@andrey-p we use an enum rather than returning the message directly so we can localise and do other stuff in the UI, but otherwise should look pretty familiar